lib/model/typing_status.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import '../api/model/events.dart';
+import 'narrow.dart';
+
+/// The model for tracking the typing status organized by narrows.
+class TypingStatus extends ChangeNotifier {
+  TypingStatus({
+    required this.selfUserId,
+    required this.typingStartedExpiryPeriod,
+  });
+
+  final int selfUserId;
+  final Duration typingStartedExpiryPeriod;
+
+  Iterable<SendableNarrow> get debugActiveNarrows => _timerMapsByNarrow.keys;
+
+  Iterable<int> typistIdsInNarrow(SendableNarrow narrow) =>
+    _timerMapsByNarrow[narrow]?.keys ?? [];
+
+  final Map<SendableNarrow, Map<int, Timer>> _timerMapsByNarrow = {};
```

Comment:
Using SendableNarrow as the key covers the narrows where typing
    notifications are supported (topics and dms).

This is good information for a comment in the tree (and then it can be omitted from the commit message). Rationale here:
https://github.com/zulip/zulip-mobile/blob/main/docs/style.md#commit-messages-vs-comments

So e.g.:
```suggestion
  // Using SendableNarrow as the key covers the narrows
  // where typing notifications are supported (topics and DMs).
  final Map<SendableNarrow, Map<int, Timer>> _timerMapsByNarrow = {};
```

(Note double-slash, not triple-slash, because that doesn't have the content of a doc comment.)

Reply:
Also updated the commit message explaining that this design was derived from the web implementation, which probably counts as some "why the new version is better than, or not worse than, the old version" information.

Reply:
Agreed, yeah that's helpful.